terraform/main.tf

```diff
@@ -21,39 +17,53 @@ module "zipkin" {
 }
 
 module "database" {
-  source         = "./modules/storage/mongo"
+  source         = "./modules/storage/database/mongo"
   image          = var.database_image
   network        = docker_network.armonik.name
   mongodb_params = var.mongodb_params
 }
 
-module "object" {
-  source  = "./modules/storage/redis"
-  image   = var.object_image
+module "object_redis" {
+  source  = "./modules/storage/object/redis"
+  count   = var.object_storage.name == "redis" ? 1 : 0
+  image   = var.object_storage.image
   network = docker_network.armonik.name
 }
 
-module "queue" {
-  source        = "./modules/queue/activemq"
+module "object_local" {
+  source = "./modules/storage/object/local"
+  count  = var.object_storage.name == "local" ? 1 : 0
+}
+
+module "queue_rabbitmq" {
+  source        = "./modules/storage/queue/rabbitmq"
+  count         = var.queue_storage.broker.name == "rabbitmq" ? 1 : 0
+  queue_storage = var.queue_storage
+  image         = var.queue_storage.broker.image
+  network       = docker_network.armonik.name
+}
+
+module "queue_activemq" {
+  source        = "./modules/storage/queue/activemq"
+  count         = var.queue_storage.broker.name == "activemq" ? 1 : 0
   queue_storage = var.queue_storage
-  image         = var.queue_image
+  image         = var.queue_storage.broker.image
   network       = docker_network.armonik.name
 }
 
+
 module "submitter" {
   source            = "./modules/submitter"
   container_name    = local.submitter.name
   core_tag          = local.submitter.tag
   docker_image      = local.submitter.image
   log_level         = local.submitter.log_level
   dev_env           = local.submitter.aspnet_core_env
-  object_storage    = local.submitter.object_storage
   network           = docker_network.armonik.name
   database_env_vars = module.database.database_env_vars
-  queue_env_vars    = module.queue.queue_env_vars
-  object_env_vars   = module.object.object_env_vars
+  queue_env_vars    = local.queue_env_vars
+  object_env_vars   = local.object_env_vars
```

Comment:
I would recommend to do the merge of the environment variables in the root module instead of the main one.
```terraform
environment = merge(local.queue_env_vars, local.object_env_vars, local.database_env_vars, local.logging_env_vars)
```